src/PostListingDisplay/components/PostListingDisplay.jsx

```diff
@@ -0,0 +1,98 @@
+import React, { Component } from 'react';
+
+import InputField from './InputField';
+
+import '../styles/PostListingDisplay.css';
+
+/**Component class that displays the input fields for a user to post a listing */
+class PostListingDisplay extends Component {
+    constructor(props){
+        super(props);
+
+        this.state = {
+            title: '',
+            description: '',
+            numRooms: '',
+            numSingles: '',
+            singlePrice: '',
+            numShared: '',
+            sharedPrice: '',
+            numBathrooms: '',
+            leaseTypes: '',
+            startDate: '',
+            endDate: '',
+            listingPrice: ''
+        };
+
+        this.handleChange = this.handleChange.bind(this);
+        this.calculateListingPrice = this.calculateListingPrice.bind(this);
+    }
+
+    handleChange(event) {
+        const value = event.target.value;
+        const name = event.target.name;
+        this.setState({ [name]: value }, () => {
+            if(name === "numSingles" || name === "singlePrice" || name === "numShared" || name === "sharedPrice"){
+                this.calculateListingPrice();
```

Comment:
I think this should either be split into two steps (pure calculation and memory update), or renamed to update so it's clear that it has side effects.

Reply:
I have split it up into updateListingPrice and calculateListingPrice. updateListingPrice calls calculateListingPrice to get what the listing price should be and then updates the state.